app/browser/reducers/tabsReducer.js

```diff
@@ -55,6 +56,16 @@ const tabsReducer = (state, action, immutableAction) => {
       break
     case appConstants.APP_TAB_UPDATED:
       state = tabState.maybeCreateTab(state, action)
+      const url = action.getIn(['tabValue', 'url'])
+      if (url) {
+        let tabValue = immutableAction.get('tabValue')
+        if (!tabValue) {
+          break
+        }
+        const bookmarked = isLocationBookmarked(state, url)
+        tabValue = tabValue.set('bookmarked', bookmarked)
+        state = tabState.updateTabValue(state, tabValue)
```

Comment:
I think I'd handle `appConstants.APP_ADD_SITE` in the `tabsReducer`, the only problem is that doesn't tell you which tab to update. You could either find the tabs with that location or pass in the info with ADD_SITE, that might be better if you can fix all the callers.

Then you could just call`updateTab` after getting the tab by tabId